contracts/NodeDelegatorETH.sol

```diff
@@ -0,0 +1,411 @@
+        emit UpdatedLRTConfig(lrtConfigAddr);
+    }
+
+    function createEigenPod() external onlyLRTManager {
+        IEigenPodManager eigenPodManager = IEigenPodManager(lrtConfig.getContract(LRTConstants.EIGEN_POD_MANAGER));
+        eigenPod = eigenPodManager.createPod();
+
+        emit EigenPodCreated(eigenPod, address(this));
+    }
+
+    /// @notice Transfers an asset back to the LRT deposit pool
+    /// @dev only supported assets can be transferred and only called by the LRT manager
+    /// @param asset the asset to transfer
+    /// @param amount the amount to transfer
+    function transferBackToLRTDepositPool(
+        address asset,
+        uint256 amount
+    )
+        external
+        whenNotPaused
+        nonReentrant
+        onlySupportedAsset(asset)
+        onlyLRTManager
+    {
+        address lrtDepositPool = lrtConfig.getContract(LRTConstants.LRT_DEPOSIT_POOL);
+
+        // Convert any ETH to WETH before transferring
+        if (asset == WETH) {
+            uint256 ethBalance = address(this).balance;
+            if (ethBalance > 0) {
+                // Convert any ETH into WETH
+                IWETH(WETH).deposit{ value: ethBalance }();
+            }
+        }
+
+        bool success = IERC20(asset).transfer(lrtDepositPool, amount);
+
+        if (!success) {
+            revert TokenTransferFailed();
+        }
+    }
+
+    /// @notice Fetches the balance of all supported assets in this contract and the underlying EigenLayer.
+    /// This includes LSTs and WETH.
+    /// @return assets a list of assets addresses
+    /// @return assetBalances the balances of the assets in the node delegator and EigenLayer
+    function getAssetBalances()
+        external
+        view
+        override
+        returns (address[] memory assets, uint256[] memory assetBalances)
+    {
+        assets = lrtConfig.getSupportedAssetList();
+        assetBalances = new uint256[](assets.length);
+
+        for (uint256 i = 0; i < assets.length;) {
+            (uint256 ndcAssets, uint256 eigenAssets) = getAssetBalance(assets[i]);
+            assetBalances[i] = ndcAssets + eigenAssets;
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @dev Returns the balance of an asset that the node delegator has deposited into an EigenLayer strategy
+    /// or native ETH staked into an EigenPod.
+    // Also needs to account for any shares pending internal withdrawal by the Prime Operator.
+    /// @param asset the token address of the asset.
+    /// WETH will include any native ETH in this contract or staked in EigenLayer.
+    /// @return ndcAssets assets lying in this NDC contract.
+    /// This includes any native ETH when the asset is WETH.
+    /// @return eigenAssets asset amount deposited in underlying EigenLayer strategy
+    /// or native ETH staked into an EigenPod.
+    function getAssetBalance(address asset) public view override returns (uint256 ndcAssets, uint256 eigenAssets) {
```

Comment:
WETH should be the only asset this contract should only return a non-zero balances for , right?

Could we just return zeros at the beginning if the token is not weth?